### ISSUE TITLE
Error: could not find template when using an enabled switch

### ISSUE THREAD

**Issue by Mo0rBy** [gruntwork-io/terratest#1328]
**Describe the bug**
I have a template yaml that uses an enabled switch in the values.yaml that determines if the template should be rendered or not:
```
{{- if .Values.security.authorizationPolicy.enabled }}
---
apiVersion: security.istio.io/v1beta1
kind: AuthorizationPolicy
metadata:
  name: {{ .Values.serviceName }}
# This continues with more yaml
{{- end }}
```

The values.yaml has `security.authorizationPolicy.enabled` which is set to false when I don't want to include an authorization-policy resource and it is set to true when I do want that resource.

When the enabled switch is set to false I get a `exit status 1; Error: could not find template templates/authorization-policy.yaml in chart`.

I was actually wanting the rendering to fail if I set the enabled switch to disabled and then I could assert that the render failed or that it produced an empty string or something similar, but this error is sort of wrong? It would indicate to me that the yaml template doesn't exist within the chart at all.

**To Reproduce**
Here's the part of my code where I set the enabled switch option and then attempt to get the Helm output for the template:
```
options := &helm.Options{ SetValues: map[string]string{ "security.authorizationPolicy.enabled": "true" }}

output := helm.RenderTemplate(GinkgoT(), options, helmChartPath, "authorization-policy", []string{"templates/authorization-policy.yaml"})
```

I imagine that this could be reproduced with any kubernetes resource, you just need to have an enabled switch at the top of the yaml template.

**Expected behavior**
When I realised why this was erroring, it sort of made sense, clearly the yaml template is not rendered because the switch is not enabled, but the error message is confusing. I think it would be better if this error message stated that the Helm template was rendered successfully, but the output was nothing, or empty, or something else indicating that the render itself worked, but no output was produced.

In the helm [`helm-unittest` plugin](https://github.com/helm-unittest/helm-unittest/tree/main) it is possible to perform this kind of assertion using the `hasDocuments` assertion (see assertion types [here](https://github.com/helm-unittest/helm-unittest/tree/main)). 

For now, I am just going to use `helm.RenderTemplateE` and assert the error, because I know for a fact that the yaml template is there and my other tests will prove this when I set the switch to enabled, but I do think that a new error should be created to more effectively state what is wrong with the test, or this functionality should be implemented that allows you to assert that no documents were generated in the Helm output.

Here is some code and terminal screenshots to help show what I am describing:
```	
Context("enabled switch", func() {
		It("should not render when set to 'disabled'", func() {
			options := &helm.Options{ SetValues: map[string]string{ "security.authorizationPolicy.enabled": "true" }} # This value is what is modified

			output := helm.RenderTemplate(GinkgoT(), options, helmChartPath, "authorization-policy", []string{"templates/authorization-policy.yaml"})
			outputNoSourceString := regexp.ReplaceAllString(output, "")

			helm.UnmarshalK8SYaml(GinkgoT(), outputNoSourceString, &authorizationPolicy)
		})
	})
``` 

When the enabled switch is true:
![Screenshot 2023-08-04 at 15 58 44](https://github.com/gruntwork-io/terratest/assets/88185573/9389f80b-de4b-4bb8-b088-fab371840a6e)

When the enabled switch is false:
![Screenshot 2023-08-04 at 16 04 08](https://github.com/gruntwork-io/terratest/assets/88185573/6ba6cf4b-4224-45c1-bfd6-e6d274d9a0d3)

**Nice to have**
- [x] Terminal output
- [x] Screenshots

**Versions**
- Terratest version: v0.43.11
- Environment details (Apple M1, macOS Ventura 13.4.1):

**Additional context**

**Comment by james03160927** [comment]
Hi @Mo0rBy, the issue seems to occur because the authorization-policy.yaml template is skipped entirely when the flag --set security.authorizationPolicy.enabled=false is used. This causes Helm to throw an error, as it interprets a template with no output as if the file does not exist or was not processed.

To address this, we can either remove the conditional check or include an else block with a placeholder comment, ensuring the template renders successfully. Since the error stems from Helm's behavior, it cannot be resolved by modifying Helm directly.

**Comment by james03160927** [comment]
Closing this issue as there's no activity on this for a while. Feel free to reopen this if you continue to face similar issue.

**Comment by Mo0rBy** [comment]
Hi @james03160927, yes I believe this is more on the Helm side to address, rather than the TerraTest side.

I've managed to work around it a long time ago by understanding why the "error" is coming up, so I just adjusted my assertions to ensure that I was getting this particular error + some other assertions, meaning I can effectively test the `enabled` flag for certain parts of my chart works correctly.

I don't think the TerraTest team should address this as, again, it's part of how Helm itself behaves. However, does this maybe deserve some documentation? Or at least an example? I'm not sure as it's a niche scenario, but then again, the `enabled` switch is a very common piece of functionality within many Helm charts and I would argue that it is an "industry standard".

Anyway, I'll leave my ranting here and there's no need to re-open the issue. Thanks!
